include/lbann/execution_algorithms/kfac/kfac_block.hpp

```diff
@@ -48,17 +60,30 @@ class kfac_block {
   kfac_block(Layer* layer,
              kfac::KFACExecutionContext* context,
              size_t layer_id,
-             size_t inverse_proc_rank)
+             size_t inverse_proc_rank,
+             bool enable_copy_errors,
+             bool enable_copy_activations,
+             int input_size,
+             int output_size)
       : m_layer(layer),
         m_layer_id(layer_id),
+        m_enable_copy_errors(enable_copy_errors),
+        m_enable_copy_activations(enable_copy_activations),
         m_inverse_proc_rank(inverse_proc_rank),
+        m_input_size(input_size),
+        m_output_size(output_size),
         m_context(context) {
     m_has_kronecker_inverse = false;
   }
   virtual ~kfac_block() = default;
 
   virtual void on_forward_prop_end(lbann_comm* comm) {}
 
+  /** @brief Get local Memory Consumption. */
+  virtual int get_local_memory_consumption() {
+    LBANN_ERROR("this function should be called via a sub-class.");
+  }
```

Comment:
```suggestion
  virtual int get_local_memory_consumption() = 0;
```
Is there a reason these aren't left pure virtual??